test/fuzz/cse.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Halide.h"
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include <time.h>
+
+using namespace Halide;
+using namespace Halide::Internal;
+using std::vector;
+
+Expr random_expr(FuzzedDataProvider &fdp, int depth, vector<Expr> &exprs) {
+    if (depth <= 0) {
+        return fdp.ConsumeIntegralInRange<int>(-5, 4);
+    }
+
+    if (!exprs.empty() && fdp.ConsumeBool()) {
+        // Reuse an existing expression
+        return exprs[fdp.ConsumeIntegralInRange<size_t>(0, exprs.size() - 1)];
```

Comment:
Looks fine, explicitly references size()-1

Reply:
Or better yet, use PickValueInArray()?

Reply:
PickValueInArray doesn't work on vectors... see [implementation](https://github.com/llvm/llvm-project/blob/main/compiler-rt/include/fuzzer/FuzzedDataProvider.h#L75-L78) although this is probably annoying enough that I might just create a PR with the LLVM project.

Reply:
(But could always just make a local `PickValueInVector()` wrapper to use in these tests, nice for clarity)